app.rb

```diff
@@ -0,0 +1,138 @@
+require_relative 'person'
+require_relative 'book'
+require_relative 'student'
+require_relative 'classroom'
+require_relative 'teacher'
+require_relative 'rental'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_books
+    @books.each_with_index do |book, index|
+      puts "#{index + 1}. Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_people
+    @people.each_with_index do |person, index|
+      puts "#{index + 1}. Name: #{person.name}, Age: #{person.age}"
+    end
+  end
+
+  def create_person
+    puts 'Enter person type (student/teacher):'
+    person_type = gets.chomp.downcase
+
+    puts 'Enter age:'
+    age = gets.chomp.to_i
+
+    if person_type == 'student'
+      puts 'Enter classroom:'
+      classroom_name = gets.chomp
+      classroom = Classroom.new(classroom_name)
+      person = Student.new(age, classroom)
+    elsif person_type == 'teacher'
+      puts 'Enter specialization:'
+      specialization = gets.chomp
+      person = Teacher.new(age, specialization)
+    else
+      puts 'Invalid person type.'
+      return
+    end
+
+    @people.push(person)
+    puts 'Person created!'
+  end
+
+  def create_book
+    puts 'Enter title:'
+    title = gets.chomp
+
+    puts 'Enter author:'
+    author = gets.chomp
+
+    book = Book.new(title, author)
+    @books.push(book)
+    puts 'Book created!'
+  end
+
+  def create_rental
+    puts 'Select a person by their index:'
+    list_people
+    person_index = gets.chomp.to_i - 1
+
+    puts 'Select a book by its index:'
+    list_books
+    book_index = gets.chomp.to_i - 1
+
+    puts 'Enter rental date (YYYY-MM-DD):'
+    date = gets.chomp
+
+    rental = @people[person_index].add_rental(@books[book_index], date)
+    @rentals.push(rental)
+    puts 'Rental created!'
+  end
+
+  def list_rentals_for_person
+    puts 'Select a person by their index:'
+    list_people
+    person_index = gets.chomp.to_i - 1
+
+    person = @people[person_index]
+    puts "Rentals for #{person.name}:"
+    person.rentals.each do |rental|
+      puts "Date: #{rental.date}, Book: #{rental.book.title}"
+    end
+  end
```

Comment:
- [ ] As stated in the project requirement Your console app should behave in the same way as it is in the example So when List all rentals for a given person id, your Console App doesn't allow the user to insert the person id as suggested in the project requirement So Kindly Fix the issues Just like the below Console App .

- A console App example for  List all rentals for a given person id just the example given
![List rental](https://github.com/BenMKT/OOP-school-library/assets/93598189/85e47604-4359-4f5a-aac4-238b1381efc3)

- Your Console App for List all rentals for a given person id behave just  like below
![List rental](https://github.com/BenMKT/OOP-school-library/assets/93598189/ae141e69-8be3-4e31-b22a-3f690634f3c5)

Reply:
- Update `list_rentals_for_person` method to resolve bug.